mm2src/coins/utxo.rs

```diff
@@ -918,7 +927,7 @@ impl<'a> UtxoConfBuilder<'a> {
 
     fn address_format(&self) -> Result<UtxoAddressFormat, String> {
         let conf = self.conf;
-        if conf["address_format"].is_null() {
+        if (conf["address_format"].is_null() || conf["address_format"]["format"] == "segwit") && !self.full_segwit() {
```

Comment:
I propose using the following code here, maybe you will improve it as I don't consider it as perfect :slightly_smiling_face: It also adds the usage of the field from the request that we discussed.
```rust
let mut format: Option<UtxoAddressFormat> =  try_s!(json::from_value(self.req["address_format"].clone());
if format.is_none() {
    format  = try_s!(json::from_value(self.conf["address_format"].clone())
}

format = format.unwrap_or(UtxoAddressFormat::Standard);

if format.is_segwit() && !self.segwit() {
     ERR!("Can not use Segwit address format for coin without segwit support")
} else {
     Ok(format)
}
```

Reply:
I don't like adding the "address_format" to the electrum request because as part of supporting other coins other than BTC (e.g. LTC) the address format will be
```
"address_format": {
   "format": "segwit",
   "hrp": "ltc"
}
```
and I don't think the hrp should be part of the request. Also, I think it's better for segwit enabling param in the request to be bool, so I can't get around using `(conf["address_format"].is_null() || conf["address_format"]["format"] == "segwit")`

Reply:
I guess that HRP is something that never (or very barely) changes so it can be simply put to the config file - there is no reason to have this prefix in the request.

If we don't put `address_format` to the `enable/electrum` then how we handle address format change? Force users and GUI developers to change coins config? If yes, it requires an MM2 restart in such a case that makes things quite complicated. Users should be able to pick the desired address format on coin activation without MM2 restart so adding the param to `enable/electrum` looks the best for this purpose.

> Also, I think it's better for segwit enabling param in the request to be bool, so I can't get around using (conf["address_format"].is_null() || conf["address_format"]["format"] == "segwit")

I mentioned above https://github.com/KomodoPlatform/atomicDEX-API/pull/953#discussion_r639623275 that we can get rid of the `force_segwit` param in the request as it makes a little sense in the end. The proposed code uses 
```rust
 fn segwit(&self) -> bool { self.conf["segwit"].as_bool().unwrap_or(false) }
 ```
not the
```rust
fn full_segwit(&self) -> bool { self.req["full_segwit"].as_bool().unwrap_or(false) }
```

Reply:
I meant by this
```
"address_format": {
   "format": "segwit",
   "hrp": "ltc"
}
```
that it's in the config file.  Maybe I didn't phrase what I meant clearly 🙂 but there will be a `segwit_address: bool` parameter in the `enable/electrum` method but it should be bool as I think making it
```
"address_format": {
   "format": "segwit",
   "hrp": "ltc"
}
```
or even `address_format": "segwit"` and not bool might not be the best way for the `enable/electrum` method

Reply:
Bech32 HRP is not necessarily should be the part of address format object, it can be something like:
```
"address_format": {
   "format": "segwit",
   "segwit_subtype": P2SH | P2WPKH | P2WSH
},
bech32_hrp: "ltc"
```

Or maybe 
```
"address_format": {
   "format": SegwitP2SH | SegwitP2WPKH | SegwitP2WSH 
},
bech32_hrp: "ltc"
```

Both variants do not contain HRP in the `address_format`, which is fine.

> there will be a segwit_address: bool parameter in the enable/electrum

I think it can not be bool because there are different Segwit address types available and users should be able to choose and switch between them.